pangeo/jobqueue.py

```diff
@@ -0,0 +1,126 @@
+from contextlib import contextmanager
+import logging
+import subprocess
+import toolz
+
+from distributed.utils import tmpfile, ignoring
+
+logger = logging.getLogger(__name__)
+
+class JobQueue(object):
+    """ Base class to launch Dask to a job queue, this class should not be used directly,
+    use inherited class appropriate for your queueing system eg PBScluster or SLURMcluster
+    """
```

Comment:
This seems like the right approach to me: a generic base class with subclasses for the specific scheduler details. 👍